File: Models/CpuFlags.cs
using System.Text.Json.Serialization;

namespace SpaceInvadersUI.Models
{
    public struct CpuFlags
    {
        [JsonInclude]
        public bool Sign;

        [JsonInclude]
        public bool Zero;

        [JsonInclude]
        public bool AuxCarry;

        [JsonInclude]
        public bool Parity;

        [JsonInclude]
        public bool Carry;
    }
}

File: src/models/CpuFlags.ts
export default class CpuFlags {
    sign: boolean;
    zero: boolean;
    auxCarry: boolean;
    parity: boolean;
    carry: boolean;

    constructor(sign: boolean, zero: boolean, auxCarry: boolean, parity: boolean, carry: boolean) {
        this.sign = sign;
        this.zero = zero;
        this.auxCarry = auxCarry;
        this.parity = parity;
        this.carry = carry;
    }
}
